src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6711,6 +6762,67 @@ components:
           items:
             type: string
 
+    SnapshotBuilderAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotBuilderRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotBuilderRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCohort'
+        conceptSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSets'
+        valueSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderFeatureValueGroup'
+      description: Specification of requested a snapshot, returned in SnapshotBuilderAccessRequest.
+
+    SnapshotBuilderCohort:
+      type: object
+      properties:
+        name:
+          type: string
+        criteriaGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCriteriaGroup'
+      description: Selection of participants.
+
+    SnapshotBuilderCriteriaGroup:
+      type: object
+      properties:
+        name:
+          type: string
+        criteria:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderAnyCriteria'
+        mustMeet:
+          type: boolean
+        meetAll:
+          type: boolean
+        count:
+          type: number
+      description: Criteria Group specifying a Cohort.
+
+    SnapshotBuilderAnyCriteria:
+      oneOf:
```

Comment:
Ah okay I see. Instead of defining AnyCriteria as its own component, I could define SnapshotBuilderCriteriaGroup.criteria as a list of type SnapshotBuilderDomainOption or SnapshotBuilderProgramDataOption

Reply:
Okay removing the explicit definition of AnyCriteria seems to get rid of the extra class definition. So now there is OneOfSnapshotBuilderCriteriaGroupCriteriaItems which has two subtypes SnapshotBuilderDomainOption or SnapshotBuilderProgramDataOption. The name is not ideal, but I wonder what you think of this approach.